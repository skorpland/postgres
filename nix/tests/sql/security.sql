-- get a list of security definer functions owned by powerbase_admin
-- this list should be vetted to ensure the functions are safe to use as security definer
select
    n.nspname, p.proname
from pg_catalog.pg_proc p
    left join pg_catalog.pg_namespace n ON n.oid = p.pronamespace
where p.proowner = (select oid from pg_catalog.pg_roles where rolname = 'powerbase_admin')
        and p.prosecdef = true
order by 1,2;
