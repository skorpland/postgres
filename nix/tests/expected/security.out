-- get a list of security definer functions owned by powerbase_admin
-- this list should be vetted to ensure the functions are safe to use as security definer
select
    n.nspname, p.proname
from pg_catalog.pg_proc p
    left join pg_catalog.pg_namespace n ON n.oid = p.pronamespace
where p.proowner = (select oid from pg_catalog.pg_roles where rolname = 'powerbase_admin')
        and p.prosecdef = true
order by 1,2;
  nspname  |            proname             
-----------+--------------------------------
 graphql   | get_schema_version
 graphql   | increment_schema_version
 pgbouncer | get_auth
 pgsodium  | disable_security_label_trigger
 pgsodium  | enable_security_label_trigger
 pgsodium  | get_key_by_id
 pgsodium  | get_key_by_name
 pgsodium  | get_named_keys
 pgsodium  | mask_role
 pgsodium  | update_mask
 public    | dblink_connect_u
 public    | dblink_connect_u
 public    | pgaudit_ddl_command_end
 public    | pgaudit_sql_drop
 public    | st_estimatedextent
 public    | st_estimatedextent
 public    | st_estimatedextent
 repack    | repack_trigger
 vault     | create_secret
 vault     | update_secret
(20 rows)

